superstring/superstring.py

```diff
@@ -108,44 +101,43 @@ def __init__(self, left, right):
     def length(self):
         return self._left.length() + self._right.length()
 
-    def characterAt(self, index):
-        leftLen = self._left.length()
-        if index < leftLen:
+    def character_at(self, index):
+        left_len = self._left.length()
+        if index < left_len:
             return self._left[index]
-        return self._right[index - leftLen]
-
-    def toPrintable(self, startIndex = None, endIndex = None):
-        if startIndex == None:
-            startIndex = 0
-        if endIndex == None:
-            endIndex = self.length()
-        leftLen = self._left.length()
-        if endIndex < leftLen:
-            return self._left.toPrintable(startIndex = startIndex, endIndex = endIndex)
-        elif startIndex > leftLen:
-            return self._right.toPrintable(startIndex = startIndex - leftLen, endIndex = endIndex - leftLen)
-        return self._left.toPrintable(startIndex = startIndex) + self._right.toPrintable(endIndex = endIndex - leftLen)
+        return self._right[index - left_len]
+
+    def to_printable(self, start_index=None, end_index=None):
+        start_index = start_index if start_index else 0
+        end_index = end_index if end_index else self.length()
+        left_len = self._left.length()
+        if end_index < left_len:
+            return self._left.to_printable(start_index=start_index, end_index=end_index)
+        elif start_index > left_len:
+            return self._right.to_printable(start_index=start_index - left_len, end_index=end_index - left_len)
+        return self._left.to_printable(start_index=start_index) + self._right.to_printable(
+            end_index=end_index - left_len)
+
 
 class SuperStringSubstring(SuperStringBase):
-    def __init__(self, base, startIndex, endIndex):
+    def __init__(self, base, start_index, end_index):
         self._base = base
-        self._startIndex = startIndex
-        self._endIndex = endIndex
+        self._start_index = start_index
+        self._end_index = end_index
 
     def length(self):
-        return self._endIndex - self._startIndex
+        return self._end_index - self._start_index
+
+    def character_at(self, index):
+        return self._base.character_at(self._start_index + index)
 
-    def characterAt(self, index):
-        return self._base.characterAt(self._startIndex + index)
+    def to_printable(self, start_index=None, end_index=None):
+        start_index = start_index if start_index else 0
+        end_index = end_index if end_index else self.length()
```

Comment:
```suggestion
        end_index = end_index if end_index is not None else self.length()
```
It will show undesirable behavior for zero otherwise.